### PR TITLE
fix(core): Remove PII from log output

### PR DESCRIPTION
**Pull request by ryanrishi**
## Summary

- Add `maskPhone`, `maskEmail`, `maskAddress` log-redaction utilities and export them from `@twilio/tac-core`
- Mask author/recipient address fields across all channels (SMS, Chat, Voice) and TAC orchestrator
- Remove full webhook payloads from debug/warn log statements
- Remove raw WebSocket message data and voice transcripts from error contexts
- Strip PII from `handleError` log output while preserving full context in customer `onError` callbacks
- Add defense-in-depth PII safety filter via Pino `hooks.logMethod` — regex-scrubs phone and email patterns from all log output before serialization, catching any PII a developer forgets to mask explicitly

## Type of Change

- [x] Bug fix
- [x] New feature

## Checklist

- [x] Tests added/updated
- [ ] Documentation updated
- [ ] Tested E2E

## Test plan

- [x] Unit tests for `maskPhone`, `maskEmail`, `maskAddress` (boundary lengths, edge cases)
- [x] Unit tests for `scrubPii` and `scrubObject` (phones, emails, nested objects, arrays, primitives)
- [x] Integration tests: Pino hook scrubs phone/email from log object fields, message strings, and nested objects
- [x] Integration test: intercept channel logger, verify `author` field is masked in `communication.created` log
- [x] Integration test: verify `payload` is not present in error context logs
- [x] Integration test: verify masked address in profile lookup warning log
- [x] Integration test: verify `author` is masked in `handle_message_ready` log

## SDK Parity

This is the **TypeScript SDK**. If this change affects shared functionality, ensure the [Python SDK](https://github.com/twilio/twilio-agent-connect-python) is updated as well.

- [ ] Change is TypeScript-specific (no Python update needed)
- [ ] Python SDK PR created: https://github.com/twilio/twilio-agent-connect-python/pull/11

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/core/src/channels/messaging.ts [hunk 1]
```diff
@@ -813,7 +815,10 @@ export abstract class MessagingChannel extends BaseChannel {
             );
           });
       }
-      this.logger.error({ err: error, to }, `Failed to initiate outbound ${channel}`);
+      this.logger.error(
+        { err: error, to: maskAddress(to) },
+        `Failed to initiate outbound ${channel}`
+      );
       this.handleError(error instanceof Error ? error : new Error(String(error)), { to });
```
[the remark this answers is not in the document]
Comment (ryanrishi) [reply packages/core/src/channels/messaging.ts:822]: handleError no longer spreads context into logs (changed in this PR — see base.ts diff). Context only flows to the customer's onError callback, where they need the raw data for their own debugging.

File: packages/core/src/channels/voice.ts [hunk 2]
```diff
@@ -638,11 +641,17 @@ export class VoiceChannel extends BaseChannel {
...
 
       return { callSid: call.sid };
     } catch (error) {
-      this.logger.error({ err: error, to: validated.to }, 'Failed to initiate outbound call');
+      this.logger.error(
+        { err: error, to: maskAddress(validated.to) },
+        'Failed to initiate outbound call'
+      );
       this.handleError(error instanceof Error ? error : new Error(String(error)), {
         to: validated.to,
```
[the remark this answers is not in the document]
Comment (ryanrishi) [reply packages/core/src/channels/voice.ts:656]: handleError no longer logs context (see base.ts change in this PR). The raw to value only reaches the customer's onError callback, which is intentional.

File: packages/core/src/channels/messaging.ts [hunk 3]
```diff
@@ -223,7 +224,7 @@ export abstract class MessagingChannel extends BaseChannel {
    * Process messaging channel webhook from Twilio Conversations Service
    */
   public async processWebhook(payload: unknown, idempotencyToken?: string): Promise<void> {
-    this.logger.debug({ operation: 'webhook_processing', payload }, 'Processing webhook');
+    this.logger.debug({ operation: 'webhook_processing' }, 'Processing webhook');
```
[the remark this answers is not in the document]
Comment (ryanrishi) [reply packages/core/src/channels/messaging.ts:228]: handleError was changed in this PR to only log { err: error } — context is never written to logs. It's passed through to the customer's onError callback where they need the full payload for debugging.

**Review by xinghaohuang91 — APPROVED** [review]
